modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -488,31 +590,92 @@ namespace ie {
 static void configureInputInfo(const IE::InputInfo::Ptr& ii, const cv::GMetaArg mm) {
     switch (mm.index()) {
         case cv::GMetaArg::index_of<cv::GMatDesc>():
-            {
-                ii->setPrecision(toIE(util::get<cv::GMatDesc>(mm).depth));
-                break;
-            }
+        {
+            ii->setPrecision(toIE(util::get<cv::GMatDesc>(mm).depth));
+            break;
+        }
         case cv::GMetaArg::index_of<cv::GFrameDesc>():
-            {
-                const auto &meta = util::get<cv::GFrameDesc>(mm);
-                switch (meta.fmt) {
-                    case cv::MediaFormat::NV12:
-                        ii->getPreProcess().setColorFormat(IE::ColorFormat::NV12);
-                        break;
-                    case cv::MediaFormat::BGR:
-                        // NB: Do nothing
-                        break;
-                    default:
-                        GAPI_Assert(false && "Unsupported media format for IE backend");
-                }
-                ii->setPrecision(toIE(CV_8U));
-                break;
+        {
+            const auto &meta = util::get<cv::GFrameDesc>(mm);
+            switch (meta.fmt) {
+                case cv::MediaFormat::NV12:
+                    ii->getPreProcess().setColorFormat(IE::ColorFormat::NV12);
+                    break;
+                case cv::MediaFormat::BGR:
+                    // NB: Do nothing
+                    break;
+                default:
+                    GAPI_Assert(false && "Unsupported media format for IE backend");
             }
+            ii->setPrecision(toIE(CV_8U));
+            break;
+        }
         default:
             util::throw_error(std::runtime_error("Unsupported input meta for IE backend"));
     }
 }
 
+struct PostOutputs
+{
+    // NB: Should be const to pass into SetCompletionCallback
+    void operator()() const
+    {
+       for (auto i : ade::util::iota(uu.params.num_out))
+       {
+            auto& out_mat = ctx->outMatR(i);
+            IE::Blob::Ptr this_blob = iec.this_request.GetBlob(uu.params.output_names[i]);
+            copyFromIE(this_blob, out_mat);
+            out.post(ctx->out(i));
+       }
+
+        sync.release_and_notify();
+    }
+
+    IECompiled                            &iec ;
+    const IEUnit                          &uu  ;
+    cv::gimpl::GIslandExecutable::IOutput &out ;
+    cv::gimpl::ie::SyncPrim               &sync;
+    std::shared_ptr<IECallContext>         ctx ;
```

Comment:
So much stuff here - I'd make it part of the context.

Reply:
Fixed